TriforkSwiftExtensions/TriforkSwiftExtensions/Array/ArrayExtension.swift

```diff
@@ -31,7 +31,52 @@ public extension Array where Element: Equatable {
             body(element, index)
         }
     }
+}
+
+extension Array where Element: Hashable {
+    
+    /// Returns a new array without duplicates.
+    func unique() -> [Element] {
+        let set = Set(self)
+        return Array(set)
+    }
+    
+    static func - (lhs: [Element], rhs: [Element]) -> [Element] {
+        var arr = Array(lhs)
+        for elem in rhs {
+            arr.removeElement(elem)
+        }
+        return arr
+    }
+}
 
+extension Array where Element == String {
+    ///Returns a concatenated string, separating the elements by the given separator and ending with a given separator before the last element.
+    ///
+    /// The following example shows how an array of strings can be joined to a
+    /// single, comma-separated and, and-separated string:
+    ///
+    /// let cast = ["Vivien", "Marlon", "Kim", "Karl"]
+    /// let list = cast.joined(by: ", ", end: " and ")
+    /// print(list)
+    /// // Prints "Vivien, Marlon, Kim and Karl"
+
+    /// - Parameter by: the initial string separator inserted between elements in the beginning of the array. *Will only be used if Array holds three or more elements.
+    /// - Parameter end: the final string separator. If the Array contains only two elements, they will be separated by the end parameter.
+    /// - Returns: A single, concatenated string.
+    func joined(by: String, end: String) -> String {
+        guard let lastElement = last else { return "" }
+        
+        return reduce("") { partialResult, element in
+            if partialResult.isEmpty {
```

Comment:
split into if's instead to reduce nesting like so
``` swift
if partialResult.isEmpty {
  return element
  }
  if element == lastElement {
     return partialResult + end + element
  }
  return partialResult + by + element
  ``` 
  but it might equally be written as 
  ```swift
  if partialResult.isEmpty {
    return element
  }
  let isLastElement = element == lastElement
  let concatElement = isLastElement.map(ifTrue: end, ifFalse: by)
  return partialResult + concatElement + element
 ```